gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopySource.java

```diff
@@ -468,7 +479,17 @@ private static void setWorkUnitWeight(WorkUnit workUnit, CopyEntity copyEntity,
       weight = ((CopyableFile) copyEntity).getOrigin().getLen();
     }
     weight = Math.max(weight, minWeight);
-    workUnit.setProp(WORK_UNIT_WEIGHT, Long.toString(weight));
+    workUnit.setProp(ConfigurationKeys.GOBBLIN_COPY_WORK_UNIT_WEIGHT, Long.toString(weight));
+  }
+
+  public static void setWorkUnitBlockLocations(WorkUnit workUnit, FileSystem sourceFs, CopyableFile file,
+      long startPos, long len) throws IOException {
+    Set<String> blockLocationsNames = Sets.newHashSet();
+    for (BlockLocation bl : sourceFs.getFileBlockLocations(file.getFileStatus(), startPos, len)) {
+      blockLocationsNames.addAll(Arrays.asList(bl.getHosts()));
+    }
+    workUnit.setProp(ConfigurationKeys.SOURCE_FILEBASED_BLOCK_LOCATIONS,
```

Comment:
We should think about where the correct location is for adding block locations. The approach you have would work, but is very Distcp-specific. Can we have the logic of fetching block locations inside the GobblinWorkUnitsInputFormat, so that other sources can benefit from it ?